### PR TITLE
doc/md: add sqlalchemy provider guide

### PR DESCRIPTION
**Pull request by ronenlu**

### CODE REVIEW

File: doc/md/guides/orms/sqlalchemy.md [hunk 1]
```diff
@@ -0,0 +1,256 @@
+---
+id: sqlalchemy
+title: Automatic migration planning for SQLAlchemy
+slug: /guides/orms/sqlalchemy
+---
+
+## TL;DR
+* [SQLAlchemy](https://www.sqlalchemy.org) is Python SQL toolkit and Object Relational Mapper.
```
Comment (hilakashai) [inline doc/md/guides/orms/sqlalchemy.md:8]:
```suggestion
* [SQLAlchemy](https://www.sqlalchemy.org) is a Python SQL toolkit and Object Relational Mapper.
```

File: doc/md/guides/orms/sqlalchemy.md [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+  for them, based on the desired state of their schema instead of crafting them by hand.
+
+## Automatic migration planning for SQLAlchemy
+
+SQLAlchemy is a popular ORM toolkit widely used in the Python community. SQLAlchemy allows users to
+manage their database schemas using its [declarative-mapping](https://docs.sqlalchemy.org/en/20/orm/declarative_tables.html)
+feature, which is usually sufficient during development and in many simple cases.
+
+However, at some point, teams need more control and decide to employ
+the [versioned migrations](/concepts/declarative-vs-versioned#versioned-migrations)  methodology,
```
Comment (hilakashai) [inline doc/md/guides/orms/sqlalchemy.md:21]:
```suggestion
the [versioned migrations](/concepts/declarative-vs-versioned#versioned-migrations) methodology,
```

File: doc/md/guides/orms/sqlalchemy.md [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+which is a more robust way to manage your database schema.
```
Comment (hilakashai) [inline doc/md/guides/orms/sqlalchemy.md:22]:
```suggestion
which is a more robust way to manage a database schema.
```

File: doc/md/guides/orms/sqlalchemy.md [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+migration scripts from the difference between the current state of the database and the desired state of the application.
+In the context of versioned migrations, the current state can be thought of as the database schema that would have been
+created by applying all previous migration scripts.
+
+However, Alembic auto generation is [not always accurate](https://alembic.sqlalchemy.org/en/latest/autogenerate.html#what-does-autogenerate-detect-and-what-does-it-not-detect)
+and cannot be relied upon to generate production-ready migration scripts without manual intervention.
+
+Atlas can automatically plan database schema migrations for developers using SQLAlchemy.
+
+The desired schema of your application can be provided to Atlas via an [External Schema Datasource](/atlas-schema/projects#data-source-external_schema)
```
Comment (hilakashai) [inline doc/md/guides/orms/sqlalchemy.md:35]:
```suggestion
The desired schema of your application can be provided to Atlas via an [External Schema Datasource](/atlas-schema/projects#data-source-external_schema),
```

File: doc/md/guides/orms/sqlalchemy.md [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+which is any program that can output a SQL schema definition to stdout.
+
+To use Atlas with SQLAlchemy, users can utilize the [SQLAlchemy Atlas Provider](https://github.com/ariga/atlas-provider-sqlalchemy),
```
Comment (hilakashai) [inline doc/md/guides/orms/sqlalchemy.md:38]:
```suggestion
To use Atlas with SQLAlchemy, users can utilize the [SQLAlchemy Atlas Provider](https://github.com/ariga/atlas-provider-sqlalchemy), which is 
```

File: doc/md/guides/orms/sqlalchemy.md [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+
+Therefore, you will need to run the provider from within your project's Python environment.
+
+:::
+
+### Standalone vs Script mode
+
+The Atlas SQLAlchemy Provider can be used in two modes:
+
+* **Standalone** - If all of your SQLAlchemy models exist in a single Module,
```
Comment (hilakashai) [inline doc/md/guides/orms/sqlalchemy.md:80]:
```suggestion
* **Standalone** - If all of your SQLAlchemy models exist in a single module,
```

File: doc/md/guides/orms/sqlalchemy.md [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+
+class Address(Base):
+    __tablename__ = "address"
+    id: Mapped[int] = mapped_column(primary_key=True)
+    email_address: Mapped[str] = mapped_column(String(30))
+    user_id: Mapped[int] = mapped_column(ForeignKey("user_account.id"))
+    user: Mapped["User"] = relationship(back_populates="addresses")
+```
+
+Using the [Standalone mode](#standalone-mode) configuration file for the provider,
```
Comment (hilakashai) [inline doc/md/guides/orms/sqlalchemy.md:185]:
```suggestion
Using the [standalone mode](#standalone-mode) configuration file for the provider,
```

File: doc/md/guides/orms/sqlalchemy.md [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+```sql
+-- Modify "user_account" table
+ALTER TABLE `user_account` ADD COLUMN `age` int NULL;
+```
+
+## Conclusion
+
+In this guide we demonstrated how projects using SQLAlchemy can use Atlas to automatically
+plan schema migrations based only on their data model. To learn more about executing
+these migrations against your production database, read the documentation for the
```
Comment (hilakashai) [inline doc/md/guides/orms/sqlalchemy.md:253]:
```suggestion
 migrations against your production database, read the documentation for the
```

File: doc/md/guides/orms/sqlalchemy.md [hunk 9]
```diff
@@ -0,0 +1,262 @@
...
+
+However, at some point, teams need more control and decide to employ
+the [versioned migrations](/concepts/declarative-vs-versioned#versioned-migrations) methodology,
+which is a more robust way to manage a database schema.
+
+The native way to manage migrations with SQLAlchemy is to use the [Alembic](https://alembic.sqlalchemy.org/en/latest/) migration tool.
+Alembic can [automatically generate](https://alembic.sqlalchemy.org/en/latest/autogenerate.html#auto-generating-migrations)
+migration scripts from the difference between the current state of the database and the desired state of the application.
+In the context of versioned migrations, the current state can be thought of as the database schema that would have been
+created by applying all previous migration scripts.
```
Comment (rotemtam) [inline doc/md/guides/orms/sqlalchemy.md:28]:
```suggestion
```
Correct me if I'm wrong, but this isn't true for alembic where the comparison is don't between a live db that's supposed to have the current schema and the desired state.

File: doc/md/guides/orms/sqlalchemy.md [hunk 10]
```diff
@@ -0,0 +1,262 @@
...
+
+However, Alembic auto generation is [not always accurate](https://alembic.sqlalchemy.org/en/latest/autogenerate.html#what-does-autogenerate-detect-and-what-does-it-not-detect)
+and cannot be relied upon to generate production-ready migration scripts without manual intervention.
```
Comment (rotemtam) [inline doc/md/guides/orms/sqlalchemy.md:31]:
```suggestion
A downside of this approach is that in order for it to work, a pre-existing database with the current version of the schema must be connected to.  In many production environments, databases should generally not be reachable from developer workstations, which means this comparison is normally done against a local copy of the database which may have undergone some changes that aren't reflected in the existing migrations.

 In addition, Alembic auto-generation is [fails to detect many kinds of changes](https://alembic.sqlalchemy.org/en/latest/autogenerate.html#what-does-autogenerate-detect-and-what-does-it-not-detect)
and cannot be relied upon to generate production-ready migration scripts without routine manual intervention.
```

File: doc/md/guides/orms/sqlalchemy.md [hunk 11]
```diff
@@ -0,0 +1,262 @@
...
+
+Atlas can automatically plan database schema migrations for developers using SQLAlchemy.
```
Comment (rotemtam) [inline doc/md/guides/orms/sqlalchemy.md:33]:
```suggestion
Atlas, on the other hand, can automatically plan database schema migrations for developers using SQLAlchemy without requiring a connection to such a database and can detect almost any kind of schema change.
```

**Review by rotemtam — COMMENTED** [review]
Please verify that my conceptual changes on the intro are correct

File: doc/md/guides/orms/sqlalchemy.md [hunk 12]
```diff
@@ -0,0 +1,269 @@
...
+* [Atlas](https://atlasgo.io) is an open-source tool for inspecting, planning, linting and
+  executing schema changes to your database.
+* Developers using SQLAlchemy can use Atlas to automatically plan schema migrations
+  for them, based on the desired state of their schema instead of crafting them by hand.
+
+## Automatic migration planning for SQLAlchemy
+
+SQLAlchemy is a popular ORM toolkit widely used in the Python community. SQLAlchemy allows users to
+describe their data model using its [declarative-mapping](https://docs.sqlalchemy.org/en/20/orm/declarative_tables.html)
+feature. To actually create the underlying tables, users can use the `Base.metadata.create_all` method which may be sufficient during development where tables can be routinely dropped and re-created.
```
Comment (rotemtam) [inline doc/md/guides/orms/sqlalchemy.md:18]:
```suggestion
feature. To actually create the underlying tables, users can use the `Base.metadata.create_all` method 
which may be sufficient during development where tables can be routinely dropped and re-created.
```

File: doc/md/guides/orms/sqlalchemy.md [hunk 13]
```diff
@@ -0,0 +1,269 @@
...
+the [versioned migrations](/concepts/declarative-vs-versioned#versioned-migrations) methodology,
+which is a more robust way to manage a database schema.
+
+The native way to manage migrations with SQLAlchemy is to use the [Alembic](https://alembic.sqlalchemy.org/en/latest/) migration tool.
+Alembic can [automatically generate](https://alembic.sqlalchemy.org/en/latest/autogenerate.html#auto-generating-migrations)
+migration scripts from the difference between the current state of the database and the desired state of the application.
+
+A downside of this approach is that in order for it to work, a pre-existing database with the current version of the schema must be connected to.
+In many production environments, databases should generally not be reachable from developer workstations,
+which means this comparison is normally done against a local copy of the database which may have undergone some changes that aren't reflected in the existing migrations.
```
Comment (rotemtam) [inline doc/md/guides/orms/sqlalchemy.md:30]:
```suggestion
which means this comparison is normally done against a local copy of the database which may have
undergone some changes that aren't reflected in the existing migrations.
```

File: doc/md/guides/orms/sqlalchemy.md [hunk 14]
```diff
@@ -0,0 +1,269 @@
...
+In addition, Alembic auto-generation is [fails to detect many kinds of changes](https://alembic.sqlalchemy.org/en/latest/autogenerate.html#what-does-autogenerate-detect-and-what-does-it-not-detect)
+and cannot be relied upon to generate production-ready migration scripts without routine manual intervention.
+
+Atlas, on the other hand, can automatically plan database schema migrations for developers using SQLAlchemy
+without requiring a connection to such a database and can detect almost any kind of schema change.
+Atlas plans migrations by calculating the diff between the _current_ state of the database,
+and its _desired_ state.
```
Comment (rotemtam) [inline doc/md/guides/orms/sqlalchemy.md:37]:
```suggestion
Atlas, on the other hand, can automatically plan database schema migrations for SQLAlchemy
without requiring a connection to such a database and can detect almost any kind of schema change.
Atlas plans migrations by calculating the diff between the _current_ state of the database,
and its _desired_ state.
```

**Review by rotemtam — APPROVED** [review]
See small suggestions

**Pull request merged**